lib/cocoapods/sandbox/headers_store.rb

```diff
@@ -85,6 +85,37 @@ def add_files(namespace, relative_header_paths, platform)
         end
       end
 
+      # Adds a header to the directory under different name.
+      #
+      # @param  [Pathname] namespace
+      #         the path where the header file should be stored relative to the
+      #         headers directory.
+      #
+      # @param  [Pathname] relative_header_path
+      #         the path of the header file relative to the Pods project
+      #         (`PODS_ROOT` variable of the xcconfigs).
+      #
+      # @param  [String] final_name
+      #         the name under which the file should be available in the
+      #         headers directory.
+      #
+      # @note   This method adds the file to the search paths.
+      #
+      # @return [Pathname]
+      #
+      def add_file(namespace, relative_header_path, final_name, platform)
```

Comment:
I need to link the file under different name and and ```add_files``` keep basename. https://github.com/CocoaPods/CocoaPods/pull/3796/files#diff-bf11041bbe5e0a8d1517490766eeaafeR82